Bug: Emulator crashes when adding workout manually

*Steps to reproduce:*
1. Choose froyoDebug as build variant for app project in Android Studio
2. Start emulator with API level 8.
3. Deploy the app from Android Studio
4. Choose the manual tab inside the Start tab
5. Set start hour, a duration, a distance and write some words for notes
6. Click save activity

*What happens:*
Emulator crashes with the following error...

```
08-15 23:06:01.832  10598-11136/org.runnerup E/AndroidRuntime﹕ FATAL EXCEPTION: AsyncTask #1
    java.lang.RuntimeException: An error occured while executing doInBackground()
            at android.os.AsyncTask$3.done(AsyncTask.java:200)
            at java.util.concurrent.FutureTask$Sync.innerSetException(FutureTask.java:273)
            at java.util.concurrent.FutureTask.setException(FutureTask.java:124)
            at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:307)
            at java.util.concurrent.FutureTask.run(FutureTask.java:137)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1068)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:561)
            at java.lang.Thread.run(Thread.java:1096)
     Caused by: java.lang.NoClassDefFoundError: java.util.ArrayDeque
            at com.squareup.okhttp.Dispatcher.<init>(Dispatcher.java:44)
            at com.squareup.okhttp.OkHttpClient.<init>(OkHttpClient.java:196)
            at com.mapbox.mapboxsdk.util.NetworkUtils.getHttpURLConnection(NetworkUtils.java:37)
            at com.mapbox.mapboxsdk.util.NetworkUtils.getHttpURLConnection(NetworkUtils.java:33)
            at com.mapbox.mapboxsdk.tileprovider.tilesource.TileJsonTileLayer$RetrieveJSONTask.doInBackground(TileJsonTileLayer.java:161)
            at com.mapbox.mapboxsdk.tileprovider.tilesource.TileJsonTileLayer$RetrieveJSONTask.doInBackground(TileJsonTileLayer.java:156)
            at android.os.AsyncTask$2.call(AsyncTask.java:185)
            at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:305)
            at java.util.concurrent.FutureTask.run(FutureTask.java:137)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1068)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:561)
            at java.lang.Thread.run(Thread.java:1096)
```

*What should happen:*
The workout should be added without crash.

Update: It's also possible to reproduce this by going into history and open an earlier workout.

--- a/app/res/layout-v14/map_area.xml
+++ b/app/res/layout-v14/map_area.xml
@@ -1,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.mapbox.mapboxsdk.views.MapView
+    android:id="@+id/mapview"
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:mapbox="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    mapbox:accessToken="@string/mapboxAccessToken"
+    mapbox:mapid="@string/mapboxMapID"
+    />

--- a/app/res/layout/detail.xml
+++ b/app/res/layout/detail.xml
@@ -16,7 +16,6 @@
   ~  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 -->
 <RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
-                xmlns:mapbox="http://schemas.android.com/apk/res-auto"
     android:id="@+id/tab1_layout"
     android:layout_width="fill_parent"
     android:layout_height="fill_parent"
@@ -189,29 +188,11 @@
                     android:weightSum="1">
                 </LinearLayout>
 
-                <LinearLayout
+                <include
                     android:id="@+id/tab_map"
+                    layout="@layout/map_area"
                     android:layout_width="match_parent"
-                    android:layout_height="match_parent" >
-
-                    <com.mapbox.mapboxsdk.views.MapView
-                            android:id="@+id/mapview"
-                            android:layout_width="fill_parent"
-                            android:layout_height="fill_parent"
-                            mapbox:mapid="@string/mapboxMapID"
-                            mapbox:accessToken="@string/mapboxAccessToken" />
-
-                    <TextView
-                        android:visibility="gone"
-                        android:layout_width="fill_parent"
-                        android:layout_height="fill_parent"
-                        android:textAppearance="?android:attr/textAppearanceLarge"
-                        android:text="@string/Map_not_available_on_Froyo"
-                        android:id="@+id/froyo_map_placeholder"
-                        android:layout_gravity="center"
-                        android:gravity="center" />
-
-                </LinearLayout>
+                    android:layout_height="match_parent"/>
             </FrameLayout>
         </LinearLayout>
     </TabHost>

--- a/app/res/layout/map_area.xml
+++ b/app/res/layout/map_area.xml
@@ -1,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<TextView
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:visibility="gone"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:textAppearance="?android:attr/textAppearanceLarge"
+    android:text="@string/Map_not_available_on_Froyo"
+    android:id="@+id/froyo_map_placeholder"
+    android:layout_gravity="center"
+    android:gravity="center"
+    />

--- a/app/src/org/runnerup/view/DetailActivity.java
+++ b/app/src/org/runnerup/view/DetailActivity.java
@@ -69,17 +69,16 @@ import com.mapbox.mapboxsdk.overlay.PathOverlay;
 import com.mapbox.mapboxsdk.views.MapView;
 
 import org.runnerup.R;
+import org.runnerup.common.util.Constants;
 import org.runnerup.content.ActivityProvider;
 import org.runnerup.content.WorkoutFileProvider;
 import org.runnerup.db.ActivityCleaner;
 import org.runnerup.db.DBHelper;
-import org.runnerup.db.entities.ActivityEntity;
 import org.runnerup.db.entities.LocationEntity;
 import org.runnerup.export.SyncManager;
 import org.runnerup.export.Synchronizer;
 import org.runnerup.export.Synchronizer.Feature;
 import org.runnerup.util.Bitfield;
-import org.runnerup.common.util.Constants;
 import org.runnerup.util.Formatter;
 import org.runnerup.util.HRZones;
 import org.runnerup.widget.TitleSpinner;
@@ -170,12 +169,6 @@ public class DetailActivity extends FragmentActivity implements Constants {
         sport = (TitleSpinner) findViewById(R.id.summary_sport);
         notes = (EditText) findViewById(R.id.notes_text);
         map = (MapView) findViewById(R.id.mapview);
-        if (android.os.Build.VERSION.SDK_INT <= android.os.Build.VERSION_CODES.FROYO) {
-            map.setVisibility(View.GONE);
-            map = null;
-            TextView placeholder = (TextView)findViewById(R.id.froyo_map_placeholder);
-            placeholder.setVisibility(View.VISIBLE);
-        }
         saveButton.setOnClickListener(saveButtonClick);
         uploadButton.setOnClickListener(uploadButtonClick);
         if (this.mode == MODE_SAVE) {

